test/api/capi/test_capi_profiling.cpp

```diff
@@ -15,7 +15,8 @@ string BuildProfilingSettingsString(const std::vector<string> &settings) {
 	return result;
 }
 
-void RetrieveAllMetrics(duckdb_profiling_info profiling_info, const std::vector<string> &settings) {
+void RetrieveAllMetrics(duckdb_profiling_info profiling_info, const std::vector<string> &settings,
+                        std::map<string, double> &cumulative_counter, std::map<string, double> &cumulative_result) {
```

Comment:
Minor nit: let's use our map and vector type here: `duckdb/common/map.hpp`, `duckdb/common/vector.hpp`. And in other places in the file.